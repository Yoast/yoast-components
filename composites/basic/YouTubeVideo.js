import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";
import IFrame from "../../composites/basic/IFrame";

const StyledIFrame = styled( IFrame )``;

/**
 * Creates a YouTubeVideo component.
 *
 * @param {Object} props The props to use for the component.
 *
 * @returns {ReactElement} The YouTubeVideo component.
 *
 * @constructor
 */
export default function YouTubeVideo( props ) {
	return (
		<StyledIFrame
			width={ props.width }
			height={ props.height }
			src={ props.src }
			title={ props.title }
		    frameBorder={ props.frameBorder }
		    allowFullScreen={ props.allowFullScreen }
		/>
	);
}

YouTubeVideo.propTypes = {
	width: PropTypes.number,
	height: PropTypes.number,
	src: PropTypes.string.isRequired,
	title: PropTypes.string.isRequired,
	frameBorder: PropTypes.number,
	allowFullScreen: PropTypes.bool,
};

YouTubeVideo.defaultProps = {
	width: 560,
	height: 315,
	frameBorder: 0,
	allowFullScreen: true,
};
