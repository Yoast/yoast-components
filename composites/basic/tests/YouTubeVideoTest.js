import React from "react";
import renderer from "react-test-renderer";

import YouTubeVideo from "../YouTubeVideo.js";

test( "the YouTubeVideo component matches the snapshot", () => {
	const component = renderer.create(
		<YouTubeVideo src="https:/www.youtube.com/embed/bIgcj_pPIbw" title="Video title"/>
	);

	let tree = component.toJSON();
	expect( tree ).toMatchSnapshot();
} );
