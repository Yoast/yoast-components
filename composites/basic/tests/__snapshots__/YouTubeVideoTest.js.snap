// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the YouTubeVideo component matches the snapshot 1`] = `
<iframe
  allowFullScreen={true}
  className="sc-bdVaJa bDWFJH"
  frameBorder={0}
  height={315}
  src="https:/www.youtube.com/embed/bIgcj_pPIbw"
  title="Video title"
  width={560}
/>
`;
